src/domain/dataverse/domain.ts

```diff
@@ -137,46 +137,44 @@ export const storeFactory = (
             T.fromIO(() => get().data),
             T.chain(data =>
               pipe(
-                data,
-                O.fromPredicate(() => !data.isLoading),
-                O.match(
-                  () => T.of(undefined),
-                  data =>
+                T.fromIO(() =>
+                  set(state => ({ data: { ...state.data, isLoading: true, error: O.none } }))
+                ),
+                T.chainFirst(() => T.fromIO(() => gateway.abortRequest())),
```

Comment:
IMHO, you don't need to call the gateway in another `IO` because you are already in.
```suggestion
                T.chainFirst(() => gateway.abortRequest()),
```

Reply:
Yes the `gateway.abortRequest()` synchronous operation is an `IO` effect. However I think it still needs `T.fromIO` to be wrapped as a Task in order to be used in the Task context.
Your suggestion was my initial attempt, but Typescript came at me with this error: `Type 'void' is not assignable to type 'Task<unknown>'.ts(2322)`

Reply:
I see the point, LGTM